README.md

```diff
@@ -7,27 +7,179 @@
+sb.glue("some_list", [1, 3, 5])
+sb.glue("some_dict", {"a": 1, "b": 2})
+sb.glue("non_json", df, 'arrow')
+```
+
+The scrapbook library can be used later to recover recorded values by reading the output notebook
+
+```python
+nb = sb.read_notebook('notebook.ipynb')
+nb.scraps
+```
+
+The storage format of the scraps is implied by the value type any registered data translators, but can be overwritten by setting the `storage` argument to a particular translator's registered name (e.g. `"json"`).
+
+This data is persisted by generating a display output with a special media type identifying the content storage format and data. These outputs are not visible in notebook rendering but still exist in the document. Scrapbook then can rehydrate the data associated with the notebook in the future by reading these cell outputs. 
+
+### highlight
+
+Display an object with the reference `name` in a retrievable manner. Unlike `glue` this is intended to generate a visible display output for notebook interfaces to render.
+
+```python
+sb.highlight("hello", "Hello World")
+sb.highlight("sharable_png", IPython.display.Image(filename=get_fixture_path("sharable.png")))
+```
+
+Like scraps these can be retrieved at a later time, though they don't cary any actual data, just the display result of some object.
+
+```python
+nb = sb.read_notebook('notebook.ipynb')
+nb.highlights()
+```
+
+More usefully, you can copy highlights from earlier executions to re-display the object in the current notebook.
+
+```python
+nb = sb.read_notebook('notebook.ipynb')
+nb.copy_highlight("sharable_png")
+```
+
+### read_notebook
+
+Reads a Notebook object loaded from the location specified at `path`. You've already seen how this function is used in the above api calls, but essentially this provides a thin wrapper over an nbformat notebook object with the ability to extract scrapbook scraps and highlights.
+
+```python
+nb = sb.read_notebook('notebook.ipynb')
+```
+
+The Notebook object also has a few legacy functions for backwards compatability with papermill's Notebook object model. As a result it can be used to read papermill execution statistics as well as scrapbook abstractions.
+
+```python
+nb.cell_timing
+nb.execution_counts
+nb.papermill_metrics
+nb.parameter_dataframe
+nb.papermill_dataframe
+```
+
+The notebook reader relies on [papermill's registered iorw](https://papermill.readthedocs.io/en/latest/reference/papermill-io.html) to enable access to a variety of sources such as -- but not limited to -- S3, Azure, and Google Cloud.
+
+### read_notebooks
+
+Reads all notebooks in a given `path` into a Scrapbook object.
+
+```python
+book = sb.read_notebooks('path/to/notebook/collection/')
+book.sorted_notebooks # get the underlying notebooks as a list
+```
+
+The Scrapbook can be used to recall all scraps across the collection:
+```python
+book.scraps
+book.combined_scraps
+```
+
+Or to collect highlights:
+
+```python
+book.highlights
+book.combined_highlights
```

Comment:
snap

collected_snaps

Reply:
scraps / snaps are maps of maps {notebook -> {name -> value}}
combined_*** were the flattened maps {name -> value}

Probably better to rename this as flat_scraps / flat_...